modules/ref_fb_module/src/statistics_fb_impl.cpp

```diff
@@ -244,15 +281,19 @@ void StatisticsFbImpl::processSignalDescriptorChanged(const DataDescriptorPtr& v
     configure();
 }
 
-void StatisticsFbImpl::processDataPacket(const DataPacketPtr& packet)
+void StatisticsFbImpl::processDataPacketTrigger(const DataPacketPtr& packet)
 {
-    if (!valid)
-        return;
-
     const auto domainPacket = packet.getDomainPacket();
-    if (!domainPacket.assigned())
-        return;
+    auto data = static_cast<Bool*>(packet.getData());
+    // Data packet from trigger only holds one value by design
+    auto triggerData = data[0];
+    // Domain packet from trigger only holds one value by design
+    auto domainStamp = static_cast<Int*>(domainPacket.getData())[0];
+    triggerHistory.addElement(triggerData, domainStamp);
```

Comment:
Maybe a check for packet sizes would be appropriate before accessing at index 0.